### PR TITLE
Fixes #12127: Add Support for Complex types of Databricks & UnityCatalog in profiler

### LINKED ISSUE #12127: Add support for Databricks complex types (arrays/structs)

### PR DESCRIPTION
**Pull request by ayush-shah**
### Describe your changes:

Fixes #12127 

I worked on adding support for databricks & UnityCatalog complex types for profiler and sampler

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: ingestion/src/metadata/profiler/interface/sqlalchemy/databricks/profiler_interface.py [hunk 1]
```diff
@@ -13,14 +13,74 @@
...
+        ).visit_column(*args, **kwargs)
+        dot_count = result.count(".")
+        if dot_count > 2:
+            splitted_result = result.split(".", 2)[-1].split(".")
+            result = ".".join(result.split(".", 2)[:-1])
+            result += "." + "`.`".join(splitted_result)
+
+        return result
+
+    HiveCompiler.visit_column = visit_column
```
Comment (pmbrull) [inline ingestion/src/metadata/profiler/interface/sqlalchemy/databricks/profiler_interface.py:47]: why are we using the hive compiler here?

**Review by pmbrull — COMMENTED** [review]
if possible, it'd be great to add a unit test around this

File: ingestion/src/metadata/profiler/interface/sqlalchemy/databricks/profiler_interface.py (hunk 1 above)
Comment (ayush-shah) [reply ingestion/src/metadata/profiler/interface/sqlalchemy/databricks/profiler_interface.py:47]: Databricks extends HiveDIalect and uses it's compiler.
In order to support Struct, we had to override the functionality.

**Review by TeddyCr — COMMENTED** [review]
Would this add support for unity catalog as well?

**Comment by ayush-shah** [comment]
> Would this add support for unity catalog as well?

Yes, as we use the sqlalchemy dialect itself to query unity catalog, it takes care of that

**Comment by ayush-shah** [comment]
> Would this add support for unity catalog as well?

Added the same in the latest PR 🙏

File: ingestion/src/metadata/profiler/interface/sqlalchemy/databricks/profiler_interface.py (hunk 1 above)
Comment (ayush-shah) [reply ingestion/src/metadata/profiler/interface/sqlalchemy/databricks/profiler_interface.py:51]: thanks, added a comment.

**Review by IceS2 — APPROVED** [review]

**Pull request merged**